crates/schema-inference/src/server.rs

```diff
@@ -188,14 +207,23 @@ async fn journal_to_shape(
 async fn fragment_to_shape(
     client: journal_client::Client,
     fragment: fragments_response::Fragment,
-    abort_signal: Receiver<()>,
-) -> Result<Option<(Shape, u64)>, InferenceError> {
+    abort_signal: Receiver<bool>,
+) -> Result<Option<ShapeAndMeta>, InferenceError> {
+    tokio::pin!(abort_signal);
+    // Let's bail early if we're already aborted
+    if *abort_signal.borrow_and_update() {
```

Comment:
uber nit: This is completely fine, but it does rely on some kinda tricky Rust behavior that's not especially well documented. This works because the compiler drops temporary values for `if` conditions immediately after evaluating the condition. But if you made a seemingly innocuous change, such as changing the `if` to a `match`, then the temporary `Ref` could easily end up living _longer_. IMO it's best to err on the side of explicitness with synchronization types, so maybe something like:

```rust
fn is_aborted(rx: &mut Receiver<bool>) -> bool {
  *rx.borrow_and_update()
}
```

Which makes it clear that the lock is definitely dropped after `is_aborted` returns.